### PR TITLE
Restrict FeedItemEventType

### PR DESCRIPTION
**Pull request by paulxuca**
https://avara.height.app/T-5148

### CODE REVIEW

File: packages/react/src/feed/useFeed.ts [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
+import { FeedEventItemType } from './FeedEventItemType';
 
 export type UseFeedArgs = PaginatedArgs<{
   observerId?: string;
   profileId: string;
   restrictEventTypesTo?: FeedEventItemType[];
 }>;
 
-export { FeedEventItemType };
+const mapRestrictEventTypesToLensTypes = (restrictEventTypesTo?: FeedEventItemType[]) => {
```
Comment (desfero) [inline packages/react/src/feed/useFeed.ts:18]: nit: wondering if we could get rid of `ifs` at all
```
const SupportedFeedEvenTypesMap: Record<FeedEventItemType, LensFeedEventItemType> = {
  [FeedEventItemType.Comment]: LensFeedEventItemType.Comment,
  [FeedEventItemType.Post]: LensFeedEventItemType.Post,
  [FeedEventItemType.Mirror]: LensFeedEventItemType.Mirror,
  [FeedEventItemType.CollectComment]: LensFeedEventItemType.CollectComment,
  [FeedEventItemType.CollectPost]: LensFeedEventItemType.CollectPost,
};

const mapRestrictEventTypesToLensTypes = (restrictEventTypesTo?: FeedEventItemType[]) => {
  return restrictEventTypesTo?.map((type) =>
    nonNullable(SupportedFeedEvenTypesMap[type], "Can't map the feed even type"),
  );
};
```

File: packages/react/src/feed/FeedEventItemType.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { FeedEventItemType as LensFeedEventItemType } from '@lens-protocol/api-bindings';
+
+export enum FeedEventItemType {
```
Comment (desfero) [inline packages/react/src/feed/FeedEventItemType.ts:3]: nit: just thinking, maybe it's not the responsibility of `@lens-protocol/react` to protect which `FeedEvenItemType` is supported but rather `@lens-protocol/react` should overwrite the `useFeedHook` to guide what can be used?

**Review by desfero — CHANGES_REQUESTED** [review]
Looks good, just small nitpicks to discuss

File: packages/react/src/feed/FeedEventItemType.ts (hunk 2 above)
Comment (desfero) [reply packages/react/src/feed/FeedEventItemType.ts:3]: after thinking a little bit more it seems not worth the effort to add more complexity to the `api-bindings`

**Review by krzysu — APPROVED** [review]

**Review by desfero — APPROVED** [review]

**Pull request merged**